src/dfx/src/lib/builders/mod.rs

```diff
@@ -336,17 +340,25 @@ pub fn get_and_write_environment_variables<'a>(
             vars.push((
                 Owned(format!(
                     "CANISTER_CANDID_PATH_{}",
-                    canister.get_name().replace('-', "_")
+                    canister.get_name().replace('-', "_").to_ascii_uppercase()
                 )),
                 Borrowed(candid_path),
             ));
         }
     }
     for canister in pool.get_canister_list() {
+        // Insert both suffixed and prefixed versions of the canister name for backwards compatibility
+        vars.push((
+            Owned(format!(
+                "{}_CANISTER_ID",
+                canister.get_name().replace('-', "_").to_ascii_uppercase(),
+            )),
+            Owned(canister.canister_id().to_text().into()),
+        ));
         vars.push((
             Owned(format!(
                 "CANISTER_ID_{}",
-                canister.get_name().replace('-', "_")
```

Comment:
Would you please add this form, in addition to the mix-cased form that we've had, so that this part is backwards-compatible?